### ISSUE TITLE
Possible Bug in modeling_moe.py: top-k Uses 10 Experts Instead of 2 (Mismatch with Paper & Eval Results)

### ISSUE THREAD

**Issue by linfeng93** [JieShibo/MoLE#7]
Hi, thank you for open-sourcing this great project!

While reviewing the MoE implementation, I found a potential issue in modeling_moe.py `line 656`:

> routing_weights, selected_experts = torch.topk(router_logits, 10, dim=-1)

According to the model description in the paper, the router should select top-2 experts, but the current implementation is selecting top-10 experts.

✅ **Experimental Verification**

I also ran evaluations using lm-eval:

With current code: the evaluation results do not match the numbers reported in the paper.

After changing to:

`torch.topk(router_logits, 2, dim=-1)`

the evaluation results become highly consistent with the paper’s reported performance.

**Comment by JieShibo** [comment]
Thank you very much for your feedback. This bug was due to our failure to backtrack the changes made in the ablation experiments. We have already updated the code.
